### PR TITLE
gh-83004: Harden msvcrt further

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-83004

### CODE REVIEW

**Comment by erlend-aasland** [comment]
The old `insert*` helpers were buggy: conversion errors were cleared, insert dict errors simply ignored (could theoretically end up overwriting the previous exception over and over again).

File: PC/msvcrtmodule.c [hunk 1]
```diff
@@ -656,14 +651,12 @@ exec_module(PyObject* m)
     if (version == NULL) {
         return -1;
     }
-    st = PyModule_AddObjectRef(m, "CRT_ASSEMBLY_VERSION", version);
+    int st = PyModule_AddObjectRef(m, "CRT_ASSEMBLY_VERSION", version);
```
Comment (kumaraditya303) [inline PC/msvcrtmodule.c:652]:
```suggestion
    st = PyModule_AddObjectRef(m, "CRT_ASSEMBLY_VERSION", version);
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
missing error label

File: PC/msvcrtmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply PC/msvcrtmodule.c:652]: I prefer to remove line 600 instead.

**Comment by erlend-aasland** [comment]
> missing error label

Thanks for the heads-up. I removed those gotos. PTAL.

**Review by kumaraditya303 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**